lib/vrl/stdlib/src/filter_array.rs

```diff
@@ -0,0 +1,141 @@
+use vrl::prelude::*;
+
+#[derive(Clone, Copy, Debug)]
+pub struct FilterArray;
+
+impl Function for FilterArray {
+    fn identifier(&self) -> &'static str {
+        "filter_array"
+    }
+
+    fn examples(&self) -> &'static [Example] {
+        &[
+            Example {
+                title: "some item match",
+                source: r#"filter_array(["foobar", "bazqux"], r'foo')"#,
+                result: Ok(r#"["foobar"]"#),
+            },
+            Example {
+                title: "no match",
+                source: r#"filter_array(["bazqux", "xyz"], r'foo')"#,
+                result: Ok(r#"[]"#),
+            },
+        ]
+    }
+
+    fn compile(&self, mut arguments: ArgumentList) -> Compiled {
+        let value = arguments.required("value");
+        let pattern = arguments.required("pattern");
+
+        Ok(Box::new(FilterArrayFn { value, pattern }))
+    }
+
+    fn parameters(&self) -> &'static [Parameter] {
+        &[
+            Parameter {
+                keyword: "value",
+                kind: kind::ARRAY,
```

Comment:
I'm wondering if we can restrict the input type further to only accept arrays of `Value::Bytes`?

Reply:
I used the the same input kind as `match_array`,  I don't think we can enforce that kind of restriction here, I guess we could do that in `resolve` but after having a quick look to existing function I did not find any previous similar implementation.